docs/docs/tooling/fuzzing.md

```diff
@@ -0,0 +1,169 @@
+---
+title: Noir Fuzzer
+description: The Noir Fuzzer is a tool that allows you to fuzz your Noir programs.
+keywords: [Fuzzing, Noir, Noir Fuzzer, Security]
+sidebar_position: 2
+---
+
+# Noir Fuzzer
+
+The Noir Fuzzer is a tool that allows you to fuzz your Noir programs. It is a type of testing tool that automatically generates and mutates test inputs to find bugs in programs. This fuzzer in particular, can automatically generate test cases for Noir programs with very little effort from the program writer.
+
+## Key Features
+- Uses coverage-guided fuzzing (tracks which parts of the program are executed)
+- Maintains a corpus of test cases that provide good coverage
+- Uses mutation-based fuzzing (modifies existing inputs to create new test cases and can use crossover)
+- Can detect discrepancies between ACIR and Brillig execution modes
+- Can explore failing programs and detect only specific failure conditions
+- Includes performance metrics and pretty printing of progress
+- Supports corpus minimization (finding a smaller set of inputs that maintain coverage), although for now only the lazy approach is implemented
+- Can be used with an oracle to perform differential fuzzing against a known good implementation in another language
+
+## Usage
+
+A simple example of a fuzzing harness is the following:
+
+```rust
+#[fuzz]
+fn fuzz_add(a: Field, b: Field) {
+    assert(a!=(b+3));
+}
+```
+
+Given a noir program, with a fuzzing harness, the fuzzer can be run with the following command:
+
+```bash
+nargo fuzz [FUZZING_HARNESS_NAME]
+```
+
+If [FUZZING_HARNESS_NAME] is given, only the fuzzing harnesses with names containing this string will be run.
+
+An example of the output of the fuzzer is shown below:
+
+![Basic fuzzer example](@site/static/img/tooling/fuzzing/basic-fuzzer-example.png)
```

Comment:
Should these images show up when we just view `fuzzing.md` for example at https://github.com/noir-lang/noir/blob/deb4edbcb4648081eb2f7d9b0a9f1e76e4e5ac00/docs/docs/tooling/fuzzing.md ?

Reply:
This is what I see: 
<img width="468" alt="Screenshot 2025-03-27 at 14 02 58" src="https://github.com/user-attachments/assets/d7200d97-af66-4af8-9ee0-f0cc899ccb09" />

Reply:
No, unfortunately they will only be visible, when the website is compiled. But you can view the website with a link from an automatically deployed one by github actions. It's above:
![image](https://github.com/user-attachments/assets/a78968c8-4271-4253-971f-8b27b8ec1603)